base/Base.jl

```diff
@@ -621,8 +621,29 @@ end
 # enable threads support
 @eval PCRE PCRE_COMPILE_LOCK = Threads.SpinLock()
 
+# stdlib shims
+let Random_PkgID = PkgId(UUID(0x9a3f8284_a2c9_5f02_9a11_845980a1fd5c), "Random")
+    global rand, randn
+
+    RANDOM_MODULE_REF = Ref{Module}()
+    # functions defined in Random
+    function rand(args...)
+        if !isassigned(RANDOM_MODULE_REF)
+            RANDOM_MODULE_REF[] = require(Random_PkgID)
+        end
```

Comment:
I'm not clear on why `RANDOM_MODULE_REF` is useful. It looks like once this ref is assigned, this `rand` method doesn't exist anymore so no more call to it can check whether the ref is assigned. So the only scenario I see where this check returns "assigned" is when a parallel call to `rand` happens before `Random` is loaded, but by the time the check is done, `Random` finished loading. As the check is the first expression of the function, it seems incredibly unlikely, and it's not too harmful in this case to call again `require`... Am i missing something?

Reply:
Methods exists in a specific world, so despite us deleting it when Random is being loaded this is still callable from a previous world.